src/core/server/rendering/types.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UnwrapPromise } from '@kbn/utility-types';
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  IKibanaResponse,
+  InternalHttpServiceSetup,
+} from '../http';
+import { IUiSettingsClient, UserProvidedValues } from '../ui_settings';
+import { Env } from '../config';
+import { PluginsServiceSetup } from '../plugins';
+import { createMetadataFactory } from './create_metadata_factory';
+
+/**
+ * @internal
+ * Just a convenience to unify places where we pass around
+ * common plugin variables.
+ */
+export type PluginVariables = Record<string, any>;
+
+/** @internal */
+export type UserVariables = Record<string, UserProvidedValues>;
+
+/** @internal */
+export type RenderingMetadata = UnwrapPromise<ReturnType<ReturnType<typeof createMetadataFactory>>>;
+
+/** @internal */
+export interface RenderingContextProviderParams {
+  request: KibanaRequest;
+  response: KibanaResponseFactory;
+  env: Env;
+  http: InternalHttpServiceSetup;
+  uiSettings: IUiSettingsClient;
+  plugins: PluginsServiceSetup;
+}
+
+/**
+ * @public
+ * Provides a client for independently rendering HTML
+ */
+export interface IRenderingContextProvider {
+  /**
+   * Generate a KibanaResponse which renders an HTML page bootstrapped
+   * with the core bundle. Intended to be returned from HTTP route handlers.
+   *
+   * * @example
+   * ```ts
+   * router.get(
+   *   { path: '/', validate: false },
+   *   (context) => context.core.rendering.render()
+   * );
+   * ```
+   *
+   * @param injectedVarsOverrides Provide optional variables to inject in the page.
+   * @param includeUserProvidedConfig Optionally disable injecting user-provided settings in the page
+   */
+  render: (
+    injectedVarsOverrides?: PluginVariables,
```

Comment:
why do we provide this API? there are no injectedVars in the NP. or we supposed to use this API for LP as well? why is `id` not configurable then?

Reply:
I removed `id` when this patch was only supposed to be for new platform, but that is being added back with the legacy implementation.